dashboard/test/models/pd/payment_terms_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'test_helper'
+
+class Pd::PaymentTermTest < ActiveSupport::TestCase
+  setup do
+    @regional_partner_1 = create :regional_partner
+    @regional_partner_2 = create :regional_partner
+
+    @workshop_1 = create :pd_workshop, num_sessions: 1, sessions_from: Date.new(2017, 1, 1), regional_partner: @regional_partner_1
+    @workshop_2 = create :pd_workshop, num_sessions: 1, sessions_from: Date.new(2017, 3, 1), regional_partner: @regional_partner_1
+    @workshop_3 = create :pd_workshop, regional_partner: @regional_partner_2
+  end
+
+  test 'throws exception if no applicable payment terms' do
+    create :pd_payment_term, regional_partner: @regional_partner_1, start_date: Date.new(2017, 4, 1)
+    [@workshop_1, @workshop_3].each do |workshop|
+      exception = assert_raises(Exception) do
+        Pd::PaymentTerm.for_workshop(workshop)
+      end
+
+      assert_equal "No payment terms were found for workshop #{workshop.id}", exception.message
+    end
+  end
+
+  test 'throws exception if too many payment terms' do
+    create :pd_payment_term, regional_partner: @regional_partner_1, start_date: Date.new(2017, 2, 1)
+    create :pd_payment_term, regional_partner: @regional_partner_1, start_date: Date.new(2017, 2, 15), end_date: Date.new(2017, 4, 1)
```

Comment:
Unless you care about specific dates, you can simply this by adding [freeze_time](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/test_helper.rb#L204-L217) to the top of this class, and then specifying dates as `Date.today`, `Date.today + 1.day`, etc. Note that many of the existing tests and factories default to that behavior (e.g. https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/factories.rb#L601-L602).

It's up to you, but I find the relative dates to be easier to read in tests. The workshops in setup can be today and 2 months from now (`Date.today` and `Date.today + 2.months`), and then it's easier to read whether these date ranges included them.